### ISSUE TITLE
Init error at the output layer

### ISSUE THREAD

**Issue by kevinsuedmersen** [kevinsuedmersen/library-neuralnetworksfromscratch#4]
Initialize the error at the output layer assuming a multi-class classification task, i.e. using the categorical cross-entropy loss and the softmax activation in the output layer
